client/signup/steps/site-topic/form.jsx

```diff
@@ -48,7 +49,7 @@ class SiteTopicForm extends Component {
 			id: verticalData.verticalId,
 			parentId: verticalData.parent,
 		} );
-	};
+	}, 777 );
```

Comment:
This neatly addresses many of the competing concerns including ensuring the debounce doesn't eventuate in false tracking data due to a delayed update. 👍

One challenge we faced was trying to debounce the visual update while decoupling that debounce from the store update, hence the `forceUpdate` in `client/signup/site-mockup/index.jsx`. 

We wanted to keep the store update immediate. The delay here creates one undesirable side-effect: calls made _within the debounce wait period_ to the `getVerticals()` [selector](https://github.com/Automattic/wp-calypso/blob/ff8c882/client/state/signup/verticals/selectors.js
) will return inaccurate results.

At present, we use `getVerticals()` to check for stored vertical results in `state.signup.verticals`, which is a bunch of key/value pairs of `search term : [ ...api_results ]`. 

If no result is found for a specific search term, we assume we've not searched or are still searching for that term. It was one way to avoid add an `isFetching` property to the state tree.

The most noticeable effect is that the search button is no longer disabled since `state.signup.steps.siteVertical.name` doesn't reflect the input field value when we check for matches in `state.signup.verticals`. [Example](https://github.com/Automattic/wp-calypso/blob/ff8c882/client/signup/steps/site-topic/form.jsx#L104).

The effect in action:

![vertical-button-is-disabled](https://user-images.githubusercontent.com/6458278/56557767-af058e00-65df-11e9-8eab-2b55c4ff7d6a.gif)

**Behaviour in production**
![vertical-button-is-enabled](https://user-images.githubusercontent.com/6458278/56557768-af9e2480-65df-11e9-9f33-43772f85732b.gif)

All this may indicate that, yes, we should in this case add something like `state.signup.vertical.isFetching` to the state to indicate the request state, or make some other adjustment. You have fresh 👀 , so it'd be great to get your input.

In the meantime, executing the tracks event in `shouldComponentUpdate` in [client/signup/site-mockup/index.jsx](https://github.com/Automattic/wp-calypso/pull/32486/files#diff-56922d013ffdb4262d58409b5171c6f0L69) might solve the issue, for example:

```javascript

   // In client/signup/site-mockup/index.jsx...
	shouldComponentUpdate( nextProps ) {
		const { siteStyle, siteType, verticalPreviewContent, verticalSlug } = nextProps;
		if ( verticalPreviewContent !== this.props.verticalPreviewContent ) {
			if (  nextProps.verticalPreviewContent ) {
                // We have new vertical preview content!
				this.props.recordTracksEvent( 'calypso_signup_site_preview_mockup_rendered', {
					site_type: siteType,
					vertical_slug: verticalSlug,
					site_style: siteStyle || 'default',
				}  );
			}
		    // Debouncing updates to the preview content
		    // prevents the flashing effect.
			this.updateDebounced();
			return false;
		}
		return true;
	}
```

Apologies for the data dump 🥔 

Let me know if I'm off track here or if you see a better way. 

Thanks!! 👍

Reply:
Thanks for finding those side-effects & for the context around them.  Very helpful.
I've updated for now to just debouncing the tracks call after the appropriate `componentDidUpdate`.

If we can avoid the `forceUpdate`, I think we'll thank ourselves later on.